### PR TITLE
Bot API integration

### PR DESCRIPTION
**Pull request by leonplondon**
# Summary
Added endpoints to create bots & make them talk.
Added DTOs for requests & validations.

# Trello Card
https://trello.com/c/03QefRws/9-integracion-con-la-api-del-bot

# Notes
- Created bean to supply instances of `WebClient` because they can be created pointing to different base urls.
- Some `@Qualifier(...)` & `@Bean(...)` where use in the source code, I was giving a try to scope/named injection.

### CODE REVIEW

File: src/main/java/wolox/reactortraining/controllers/BotController.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+public class BotController {
+
+    @Autowired
+    private IBotService botService;
+
+    @PostMapping
+    public Mono<BotDto> create(@Valid @RequestBody BotDto botDto) {
+        return Mono
+            .just(botDto)
+            .flatMap(bot -> botService.create(botDto))
```
Comment (yanzacd) [inline src/main/java/wolox/reactortraining/controllers/BotController.java]: You should send bot to the botService

File: src/main/java/wolox/reactortraining/beans/WebClientBean.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+public class WebClientBean {
+
+    @Value("${api.bots.uri}")
+    private String botsUri;
+
+    @Bean("bot-api")
+    public WebClient botApiClient() {
+        return WebClient.create(botsUri);
+    }
+}
```
Comment (Jhovannycanaspino) [inline src/main/java/wolox/reactortraining/beans/WebClientBean.java:18]: Why create a bean for the webClient?, can be simplified by injecting it into the service builder

File: src/main/java/wolox/reactortraining/controllers/BotController.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    @GetMapping
+    public Mono<BotResponse> talk(@NotEmpty @RequestParam("name") String name,
+        @RequestParam(value = "length", defaultValue = "60") Integer length) {
+        return botService
+            .talk(name, length)
+            .log();
+    }
+}
```
Comment (Jhovannycanaspino) [inline src/main/java/wolox/reactortraining/controllers/BotController.java:42]: the default size is 20

Comment (Jhovannycanaspino) [reply src/main/java/wolox/reactortraining/controllers/BotController.java:42]: placing the uris in the @Mapping, helps their readability

**Review by Jhovannycanaspino — COMMENTED** [review]
very good implementation in the construction of the requests the uris

File: src/main/java/wolox/reactortraining/beans/WebClientBean.java (hunk 2 above)
Comment (leonplondon) [reply src/main/java/wolox/reactortraining/beans/WebClientBean.java:18]: As commented in the PR description, I opted to use scope-base injection to try out the feature.

**Review by Jhovannycanaspino — APPROVED** [review]

**Review by yanzacd — APPROVED** [review]

**Pull request merged**